### PR TITLE
Integrated services endpoints

### PR DESCRIPTION
**Pull request by siliconbrain**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | yes
| Related tickets | -
| License         | Apache 2.0

### What's in this PR?
Rename features to integrated services in API endpoints/transport layer.

### Why?
Make API consistent with new naming (features → integrated services)

### Checklist
- [x] Implementation tested (with at least one cloud provider)
- [x] OpenAPI and Postman files updated (if needed)
- [x] User guide and development docs updated (if needed)

### CODE REVIEW

**Comment by sagikazarmark** [comment]
The `featureName` variable name should probably be `serviceName` even in legacy routes

File: internal/integratedservices/integratedservicedriver/transport_http.go [hunk 1]
```diff
@@ -45,28 +48,30 @@ func RegisterHTTPHandlers(endpoints Endpoints, router *mux.Router, errorHandler
 		options...,
 	))
 
-	router.Methods(http.MethodGet).Path("/{featureName}").Handler(kithttp.NewServer(
+	serviceRouter := router.Path(fmt.Sprintf("/{%s}", integratedServiceNameParamKey)).Subrouter()
```
Comment (sagikazarmark) [inline internal/integratedservices/integratedservicedriver/transport_http.go]: I'd rather stick to a single `serviceName` key and use that even in the legacy routes. It does not break backwards compatibility.

File: cmd/pipeline/main.go [hunk 2]
```diff
@@ -810,16 +810,33 @@ func main() {
...
+					integratedservicedriver.RegisterHTTPHandlers(
+						endpoints,
+						clusterRouter.PathPrefix("/features").Subrouter(),
+						errorHandler,
+						kitxhttp.ServerOptions(httpServerOptions),
+						kithttp.ServerErrorHandler(emperror.MakeContextAware(errorHandler)),
+					)
+
+					cRouter.Any("/features", gin.WrapH(router))
+					cRouter.Any("/features/:featureName", gin.WrapH(router)) // NOTE: parameter name does not matter here
```
Comment (sagikazarmark) [inline cmd/pipeline/main.go]: I think it does and it should.

File: cmd/pipeline/main.go [hunk 3]
```diff
@@ -810,16 +810,33 @@ func main() {
...
+						clusterRouter.PathPrefix("/services").Subrouter(),
+						errorHandler,
+						kitxhttp.ServerOptions(httpServerOptions),
+						kithttp.ServerErrorHandler(emperror.MakeContextAware(errorHandler)),
+					)
 
-				cRouter.Any("/features", gin.WrapH(router))
-				cRouter.Any("/features/:featureName", gin.WrapH(router))
+					cRouter.Any("/services", gin.WrapH(router))
+					cRouter.Any("/services/:serviceName", gin.WrapH(router)) // NOTE: parameter name does not matter here
```
Comment (sagikazarmark) [inline cmd/pipeline/main.go]: I think it does, and it should. See my comments bellow.

File: cmd/pipeline/main.go (hunk 2 above)
Comment (siliconbrain) [reply cmd/pipeline/main.go]: Now it shouldn't anymore.

File: cmd/pipeline/main.go (hunk 3 above)
Comment (siliconbrain) [reply cmd/pipeline/main.go]: Now it shouldn't anymore.

File: internal/integratedservices/integratedservicedriver/transport_http.go (hunk 1 above)
Comment (sagikazarmark) [inline internal/integratedservices/integratedservicedriver/transport_http.go]:
```suggestion
	router = router.Path(fmt.Sprintf("/{%s}", integratedServiceNameParamKey)).Subrouter()
```

File: internal/integratedservices/integratedservicedriver/transport_http.go [hunk 4]
```diff
@@ -45,28 +48,30 @@ func RegisterHTTPHandlers(endpoints Endpoints, router *mux.Router, errorHandler
...
+
+	serviceRouter.Methods(http.MethodGet).Handler(kithttp.NewServer(
```
Comment (sagikazarmark) [inline internal/integratedservices/integratedservicedriver/transport_http.go]:
```suggestion
	router.Methods(http.MethodGet).Handler(kithttp.NewServer(
```

File: internal/integratedservices/integratedservicedriver/transport_http.go [hunk 5]
```diff
@@ -45,28 +48,30 @@ func RegisterHTTPHandlers(endpoints Endpoints, router *mux.Router, errorHandler
...
 		endpoints.Details,
 		decodeIntegratedServiceDetailsRequest,
 		encodeIntegratedServiceDetailsResponse,
 		options...,
 	))
 
-	router.Methods(http.MethodPost).Path("/{featureName}").Handler(kithttp.NewServer(
+	serviceRouter.Methods(http.MethodPost).Handler(kithttp.NewServer(
```
Comment (sagikazarmark) [inline internal/integratedservices/integratedservicedriver/transport_http.go]:
```suggestion
	router.Methods(http.MethodPost).Handler(kithttp.NewServer(
```

File: internal/integratedservices/integratedservicedriver/transport_http.go [hunk 6]
```diff
@@ -45,28 +48,30 @@ func RegisterHTTPHandlers(endpoints Endpoints, router *mux.Router, errorHandler
...
 		endpoints.Activate,
 		decodeActivateIntegratedServiceRequest,
 		encodeActivateIntegratedServicesResponse,
 		options...,
 	))
 
-	router.Methods(http.MethodPut).Path("/{featureName}").Handler(kithttp.NewServer(
+	serviceRouter.Methods(http.MethodPut).Handler(kithttp.NewServer(
```
Comment (sagikazarmark) [inline internal/integratedservices/integratedservicedriver/transport_http.go]:
```suggestion
	router.Methods(http.MethodPut).Handler(kithttp.NewServer(
```

File: internal/integratedservices/integratedservicedriver/transport_http.go [hunk 7]
```diff
@@ -45,28 +48,30 @@ func RegisterHTTPHandlers(endpoints Endpoints, router *mux.Router, errorHandler
...
 		endpoints.Update,
 		decodeUpdateIntegratedServicesRequest,
 		encodeUpdateIntegratedServiceResponse,
 		options...,
 	))
 
-	router.Methods(http.MethodDelete).Path("/{featureName}").Handler(kithttp.NewServer(
+	serviceRouter.Methods(http.MethodDelete).Handler(kithttp.NewServer(
```
Comment (sagikazarmark) [inline internal/integratedservices/integratedservicedriver/transport_http.go]:
```suggestion
	router.Methods(http.MethodDelete).Handler(kithttp.NewServer(
```

**Pull request merged**